lib/matplotlib/axes/_axes.py

```diff
@@ -2665,18 +2677,26 @@ def sign(x):
                 xy = endpt, yc
 
             if orientation == "vertical":
-                xytext = 0, sign(dat) * padding
+                y_direction = -1 if y_inverted else 1
+                xytext = 0, y_direction * sign(dat) * padding
             else:
-                xytext = sign(dat) * padding, 0
+                x_direction = -1 if x_inverted else 1
+                xytext = x_direction * sign(dat) * padding, 0
 
             if label_type == "center":
                 ha, va = "center", "center"
             elif label_type == "edge":
                 if orientation == "vertical":
                     ha = 'center'
-                    va = 'top' if dat < 0 else 'bottom'  # also handles NaN
+                    if y_inverted:
+                        va = 'top' if dat > 0 else 'bottom'  # also handles NaN
+                    else:
+                        va = 'top' if dat < 0 else 'bottom'  # also handles NaN
```

Comment:
I'm not entirely certain this is correct, because there are no tests with NaN on an inverted axis.

Reply:
@Valdes-Tresanco-MS   di this get addressed?  If so, please move back out of draft.   Thanks!

Reply:
There was a new test added, but the problem is that the result that is being checked is the same as what occurs when you run the code on `main`. So I'm not certain if the test is asserting that this PR changes things correctly (and the test just happens to match `main` coincidentally), or the test is asserting that this PR doesn't break how things currently work (and the match is intentional).

Reply:
So what I guess I'm saying is it would be better if the test were paramterized across all the 4 cases here.